pkg/types/ovirt/validation/machinepool.go

```diff
@@ -1,12 +1,48 @@
 package validation
 
 import (
+	"fmt"
+
 	"k8s.io/apimachinery/pkg/util/validation/field"
 
 	"github.com/openshift/installer/pkg/types/ovirt"
+	"github.com/openshift/installer/pkg/validate"
 )
 
 // ValidateMachinePool checks that the specified machine pool is valid.
 func ValidateMachinePool(p *ovirt.MachinePool, fldPath *field.Path) field.ErrorList {
-	return field.ErrorList{}
+	allErrs := field.ErrorList{}
+
+	if p.CPU != nil {
+		if p.CPU.Cores <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("cores"), p.CPU.Cores, "CPU cores must be positive"))
+		}
+		if p.CPU.Sockets <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("sockets"), p.CPU.Sockets, "CPU sockets must be positive"))
+		}
+	}
+
+	if p.VMType != "" && !p.VMType.ValidValue() {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("vmType"), p.VMType, fmt.Sprintf("VM type must be one of %s", p.VMType.Values())))
+	}
+
```

Comment:
there is not validation that ensures that the VMType and CPU are not set simultaneously??

Reply:
nvm, i see it is here https://github.com/openshift/installer/pull/3399/commits/e002b1a403f43ce70fc6cf2a035fe16ae64f8663#diff-bcbb520f02f11439425a1a60d9d5df3cR30-R32